File: hmode/shared/golden-repos/python-cli/src/commands/process.py
"""Process command - example async command with progress."""

from pathlib import Path
from time import sleep

import click
from rich.console import Console
from rich.progress import Progress, SpinnerColumn, TextColumn

from ..logging_config import get_logger

console = Console()
log = get_logger("process")


@click.command()
@click.argument("files", nargs=-1, required=True, type=click.Path(exists=False))
@click.option("-d", "--dry-run", is_flag=True, help="Show what would be processed")
@click.pass_context
def process(ctx: click.Context, files: tuple[str, ...], dry_run: bool) -> None:
    """Process FILES with progress indication."""
    log.debug("Starting process command", files=files, dry_run=dry_run)

    if dry_run:
        console.print("[yellow]Dry run mode - no changes will be made[/yellow]")
        for file in files:
            console.print(f"  Would process: {file}")
        return

    with Progress(
        SpinnerColumn(),
        TextColumn("[progress.description]{task.description}"),
        console=console,
    ) as progress:
        task = progress.add_task("Processing files...", total=len(files))

        for i, file in enumerate(files):
            progress.update(task, description=f"Processing {file} ({i + 1}/{len(files)})")

            # Simulate processing
            sleep(0.5)
            log.info("Processed file", file=file)

            progress.advance(task)

    console.print(f"[green]Successfully processed {len(files)} files[/green]")

File: shared/golden-repos/typescript-nodejs-cli/src/commands/process.ts
/**
 * Process command - example async command with spinner.
 */

import { Command } from 'commander';
import ora from 'ora';
import chalk from 'chalk';
import { getLogger } from '../logger.js';

const log = getLogger('process');

export const processCommand = new Command('process')
  .description('Process files with progress indication')
  .argument('<files...>', 'Files to process')
  .option('-d, --dry-run', 'Show what would be processed without doing it')
  .action(async (files: string[], options: { dryRun?: boolean }) => {
    log.debug({ files, options }, 'Starting process command');

    if (options.dryRun) {
      console.log(chalk.yellow('Dry run mode - no changes will be made'));
      files.forEach((file) => {
        console.log(`  Would process: ${file}`);
      });
      return;
    }

    const spinner = ora('Processing files...').start();

    try {
      for (let i = 0; i < files.length; i++) {
        const file = files[i];
        spinner.text = `Processing ${file} (${i + 1}/${files.length})`;

        // Simulate processing
        await new Promise((resolve) => setTimeout(resolve, 500));

        log.info({ file }, 'Processed file');
      }

      spinner.succeed(chalk.green(`Successfully processed ${files.length} files`));
    } catch (error) {
      spinner.fail(chalk.red('Processing failed'));
      log.error({ error }, 'Process command failed');
      process.exit(1);
    }
  });
